src/quorum.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+pub mod majority;
+
+use std::collections::HashMap;
+use std::fmt::{self, Debug, Display, Formatter};
+
+/// VoteResult indicates the outcome of a vote.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum VoteResult {
+    /// Pending indicates that the decision of the vote depends on future
+    /// votes, i.e. neither "yes" or "no" has reached quorum yet.
+    Pending,
+    // Lost indicates that the quorum has voted "no".
+    Lost,
+    // Won indicates that the quorum has voted "yes".
+    Won,
+}
+
+/// Index is a Raft log position.
+#[derive(Default, Clone, Copy)]
+pub struct Index {
+    pub index: u64,
+    pub group_id: u64,
+}
+
+impl Display for Index {
+    #[inline]
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if self.index != u64::MAX {
+            write!(f, "[{}]{}", self.group_id, self.index)
+        } else {
+            write!(f, "[{}]∞", self.group_id)
+        }
+    }
+}
+
+impl Debug for Index {
+    #[inline]
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        Display::fmt(self, f)
+    }
+}
+
+pub trait AckedIndexer {
```

Comment:
It could be better to add some comments for this public trait

Reply:
It's not a public trait. It can only be accessed by the crate. We have `#![deny(missing_docs)]` check.